internal/transport/http2_server.go

```diff
@@ -1318,14 +1319,15 @@ func (t *http2Server) RemoteAddr() net.Addr {
 	return t.remoteAddr
 }
 
-func (t *http2Server) Drain() {
+func (t *http2Server) Drain(debugData ...string) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
 	if t.drainEvent != nil {
 		return
 	}
 	t.drainEvent = grpcsync.NewEvent()
-	t.controlBuf.put(&goAway{code: http2.ErrCodeNo, debugData: []byte{}, headsUp: true})
+	data := strings.Join(debugData, " ")
```

Comment:
The caller can send any string value as debug data. Accepting a string as an argument here and passing it as is to the controlBuf seems the most appropriate thing to do.